st2common/st2common/runners/utils.py

```diff
@@ -175,3 +178,12 @@ def invoke_post_run(liveaction_db, action_db=None):
 
     # Invoke the post_run method.
     runner.post_run(liveaction_db.status, liveaction_db.result)
+
+
+def get_file_params_path(execution_id):
+    """Returns file path for file params
+
+    :param execution_id: Execution ID of action execution.
+    :type execution_id: ``str``
+    """
+    return os.path.join(PARAM_FILE_LOCATION, PARAM_FILE_PREPEND + execution_id)
```

Comment:
To be on the safe side, we should probably use `tempfile.mkstemp` (we can still use suffix and prefix parameters of that function).

Reply:
This should also take care of permissions for us, but yeah, we still need test cases for that https://docs.python.org/2/library/tempfile.html#tempfile.mkstemp

Reply:
+1 may secure us from the issues like https://cwe.mitre.org/data/definitions/377.html and others.